core/common/src/main/java/alluxio/security/authentication/ChannelAuthenticator.java

```diff
@@ -115,35 +115,41 @@ public Channel authenticate(ManagedChannel managedChannel, AlluxioConfiguration
       return managedChannel;
     }
 
-    // Create a channel for talking with target host's authentication service.
-    // Create SaslClient for authentication based on provided credentials.
-    SaslClient saslClient;
-    if (mUseSubject) {
-      saslClient =
-          SaslParticipantProvider.Factory.create(mAuthType).createSaslClient(mParentSubject, conf);
-    } else {
-      saslClient = SaslParticipantProvider.Factory.create(mAuthType).createSaslClient(mUserName,
-          mPassword, mImpersonationUser);
+    try {
+      // Create a channel for talking with target host's authentication service.
+      // Create SaslClient for authentication based on provided credentials.
+      SaslClient saslClient;
+      if (mUseSubject) {
+        saslClient = SaslParticipantProvider.Factory.create(mAuthType)
+            .createSaslClient(mParentSubject, conf);
+      } else {
+        saslClient = SaslParticipantProvider.Factory.create(mAuthType).createSaslClient(mUserName,
+            mPassword, mImpersonationUser);
+      }
+
+      // Create authentication scheme specific handshake handler.
+      SaslHandshakeClientHandler handshakeClient =
+          SaslHandshakeClientHandler.Factory.create(mAuthType, saslClient);
+      // Create driver for driving sasl traffic from client side.
+      SaslStreamClientDriver clientDriver =
+          new SaslStreamClientDriver(handshakeClient, mGrpcAuthTimeoutMs);
+      // Start authentication call with the service and update the client driver.
+      StreamObserver<SaslMessage> requestObserver =
+          SaslAuthenticationServiceGrpc.newStub(managedChannel).authenticate(clientDriver);
+      clientDriver.setServerObserver(requestObserver);
+      // Start authentication traffic with the target.
+      clientDriver.start(mChannelId.toString());
+      // Authentication succeeded!
+      // Attach scheme specific interceptors to the channel.
+
+      Channel authenticatedChannel =
+          ClientInterceptors.intercept(managedChannel, getInterceptors(saslClient));
+      return authenticatedChannel;
+    } catch (Exception exc) {
+      LOG.error("Channel authentication failed. ChannelId:{}, AuthType:{}, Target:{}, Error:{}",
+          mChannelId, mAuthType, managedChannel.authority(), exc.getMessage());
+      throw exc;
```

Comment:
this will likely cause the same exception to be logged multiple times, since the place that catches the exception will also log it. Can we instead attach the context to the error message, e.g. 
```java
message = String.format("Channel authentication failed. ChannelId:%s, AuthType:%s, Target:%s, Error:%s", ...);
if (exc instanceof AlluxioStatusException) {
  throw new AlluxioStatusException(((AlluxioStatusException) exc).getStatus(), message, exc);
} else {
  throw new UnknownException(message, exc);
}
```